src/algebra/regular/basic.lean

```diff
@@ -47,6 +47,10 @@ protected lemma mul_le_cancellable.is_left_regular [partial_order R] {a : R}
   (ha : mul_le_cancellable a) : is_left_regular a :=
 ha.injective
 
+lemma commute.is_regular_iff {a : R} (ca : ∀ b, commute a b) :
+  is_regular a ↔ is_left_regular a :=
+⟨λ h, h.left, λ h, ⟨h, λ x y xy, h (by simpa only [(ca _).eq] using xy)⟩⟩
```

Comment:
Is it worth splitting this into two lemmas?
```suggestion
lemma is_left_regular.right_of_commute {a : R} (ca : ∀ b, commute a b)
  (h : is_left_regular a) : is_right_regular a :=
λ x y xy, h $ (ca x).trans $ xy.trans $ (ca y).symm

lemma commute.is_regular_iff {a : R} (ca : ∀ b, commute a b) :
  is_regular a ↔ is_left_regular a :=
⟨λ h, h.left, λ h, ⟨h, h.right_of_commute ca⟩⟩
```

Reply:
Sure, seems reasonable!  I deal very rarely with non-commutative rings, so I do not have a good intuition for which asymmetric lemmas might be more useful.